components/gantry/singleaxis/singleaxis.go

```diff
@@ -456,19 +469,34 @@ func (g *singleAxis) Lengths(ctx context.Context, extra map[string]interface{})
 }
 
 // MoveToPosition moves along an axis using inputs in millimeters.
-func (g *singleAxis) MoveToPosition(ctx context.Context, positions []float64, extra map[string]interface{}) error {
+func (g *singleAxis) MoveToPosition(ctx context.Context, positions, speeds []float64, extra map[string]interface{}) error {
 	ctx, done := g.opMgr.New(ctx)
 	defer done()
 
 	if len(positions) != 1 {
-		return fmt.Errorf("MoveToPosition needs 1 position to move, got: %v", len(positions))
+		return fmt.Errorf("single-axis MoveToPosition needs 1 position to move, got: %v", len(positions))
+	}
+
+	if len(speeds) != 1 {
+		return fmt.Errorf("single-axis MoveToPosition needs 1 speed to move, got: %v", len(speeds))
 	}
 
 	if positions[0] < 0 || positions[0] > g.lengthMm {
 		return fmt.Errorf("out of range (%.2f) min: 0 max: %.2f", positions[0], g.lengthMm)
 	}
 
+	switch s := speeds[0]; {
+	case s < 0:
+		speeds[0] = g.rpm
+	case s == 0:
```

Comment:
if we are doing a too slow comparison, since it is a float
```suggestion
	case rdkutils.Float64AlmostEqual(math.Abs(s), 0.0,0.1):
```